lib/phoenix/channel/transport.ex

```diff
@@ -112,65 +118,61 @@ defmodule Phoenix.Channel.Transport do
   The server will respond to heartbeats with the same message
   """
   def dispatch(socket, "phoenix", "heartbeat", _msg) do
-    send socket.pid, {:socket_reply, %Message{topic: "phoenix", event: "heartbeat", payload: %{}}}
+    socket
+    |> Socket.Server.dispatch_reply(%Message{topic: "phoenix", event: "heartbeat", payload: %{}})
```

Comment:
Yes exactly. In this context `socket.pid` is the transport's pid, which we will still need a reference to and it can stay the simple `send` we have here.